GVFS/GVFS.Common/Database/PlaceholderTable.cs

```diff
@@ -175,6 +175,57 @@ public void Remove(string path)
             }
         }
 
+        public int GetFilePlaceholdersCount()
+        {
+            try
+            {
+                using (IDbConnection connection = this.connectionPool.GetConnection())
+                using (IDbCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = $"SELECT count(path) FROM Placeholder WHERE pathType = {(int)PlaceholderData.PlaceholderType.File};";

```

Comment:
I think what you have is probably the way to go.

There could be another option maybe to do:
    SELECT count(path) FROM Placeholder GROUP BY pathType
That should negate the need for 3 separate queries.